.tekton/determine-image-digest.yaml

```
@@ -0,0 +1,37 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: determine-image-digest
+  namespace: rh-acs-tenant
+spec:
+  description: Determines image digests.
+  params:
+  - name: IMAGE_REPOSITORY
+    description: Image repository.
+    type: string
+  - name: IMAGE_TAG
+    description: Image tag.
+    type: string
+  results:
+  - name: IMAGE_DIGEST
+    description: Image digest.
+  workspaces:
+  - name: pull-secret
+    readOnly: true
+  steps:
+  - name: determine-image-digest
+    image: registry.access.redhat.com/ubi9/skopeo@sha256:a096332b3a6665b1f1e0b34980406649920cf4babba80aff24a715b76f3f62f3
+    env:
+    - name: REGISTRY_AUTH_FILE
+      value: "$(workspaces.pull-secret.path)/.dockerconfigjson"
+    script: |
+      #!/usr/bin/env bash
+      set -euo pipefail
+      image_ref="docker://$(params.IMAGE_REPOSITORY):$(params.IMAGE_TAG)"
+      digest=$(skopeo inspect \
+        --override-os linux \
+        --override-arch amd64 \
+        --format '{{.Digest}}' \
+        --no-tags \
+        "${image_ref}")
+      echo -n "$digest" | tee "$(results.IMAGE_DIGEST.path)"
```

[nit] There's no need to store the digest in a variable, it can be passed directly to the results.
```suggestion
      skopeo inspect \
        --override-os linux \
        --override-arch amd64 \
        --format '{{.Digest}}' \
        --no-tags \
        "${image_ref}" | tee "$(results.IMAGE_DIGEST.path)"
```

---

I don't have strong feelings here, but split that into two steps for the purpose of improved readability.